File: 2024/April/4-24 1137 Nth Tribonacci Number/Tribonacci.java
class Solution {
    public int tribonacci(int n) {
        if(n == 0) {
            return 0;
        }
        if(n == 1) {
            return 1;
        }
        if(n == 2) {
            return 1;
        }

        int n0 = 0;
        int n1 = 1;
        int n2 = 1;
        int n3 = 0;
        for(int i = 3; i <= n; i++) {
            n3 = n0 + n1 + n2;
            n0 = n1;
            n1 = n2;
            n2 = n3; 
        }

        return n3;
    }
}

File: 2024/April/4-24 1137 Nth Tribonacci Number/Tribonacci.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0
        if n == 1:
            return 1
        if n == 2:
            return 1

        n0 = 0
        n1 = 1
        n2 = 1
        n3 = 0
        for i in range(3, n + 1, 1):
            n3 = n0 + n1 + n2
            n0 = n1
            n1 = n2
            n2 = n3
        
        return n3